### DISCUSSION TITLE
How do I change the display language?

### DISCUSSION THREAD

**Discussion by coolvitto** [dynobo/normcap#673] [Q&A] [answered]
I created a Japanese language file (messages.po), but I don't know how to change the language used.
How do I change it?
[ja_JP.zip](https://github.com/user-attachments/files/17701905/ja_JP.zip)

**Comment by dynobo** [comment]
The UI language defaults to your system's language (or English, if NormCap does not support the system language yet). It is also possible to [start NormCap with the `LANGUAGE`/`LANG` env variable](https://github.com/dynobo/normcap/tree/main/normcap/resources/locales/README.md#test-existing-translations-in-normcap) to force as specific language (e.g. for testing).

However, you are a talking about a _new_ language, as Japanese has not been added to NormCap, yet. 

Unfortunately, the process of testing such a new language file is more complicated than just creating a `.po`-file, as it first needs to be compiled. The required steps [are described here](https://github.com/dynobo/normcap/tree/main/normcap/resources/locales/README.md#add-new-translation).

If you like, I can do those steps for you and include your translation into the next release of NormCap. Then you (or others) can test it in the application and submit corrections, if necessary. What do you think?

**Comment by coolvitto** [comment]
I read through the URLs you introduced, but I didn't really understand them.
I'll wait until they support changing the display language.

**Reply by dynobo** [reply]
Please be aware, that there are no plans to add _additional_ options for switching display language. (As said, it's already possible by changing your system's language or starting NormCap with `LANG`/`LANGUAGE` option.)

Would you allow me to add your translation `ja_JP.po` from the zip file you linked, anyway? I think other users might benefit from it.

**Comment by coolvitto** [comment]
I don't really understand how to start it with options.
How do I start it with a command in the terminal?

Please feel free to add translation files.

**Reply by dynobo** [reply]
Which operating system do you use? macOS? Windows?

**Comment by coolvitto** [comment]
Windows

**Comment by dynobo** [comment, marked as answer]
On Windows, you have to open the CMD terminal and run two commands:

1. First make sure that NormCap is closed (not running in system tray`
2. Then run: `set LANG=de_DE` (or any other supported language)
3. Followed by:  `%LOCALAPPDATA%\Programs\NormCap\NormCap.exe`

The change is only temporary (e.g. for testing). To make it permanent you have to set "LANG" in the environment variables of Windows system settings.

**Comment by coolvitto** [comment]
Wow!
It worked!!
Thank you so much!!!!!
